lib/paypal_express/api.rb

```diff
@@ -227,12 +266,23 @@ def get_active_merchant_module
 
       private
 
+      def find_payer_api(token, kb_account_id, kb_tenant_id, options = {})
+        # Go to Paypal to get the Payer id (GetExpressCheckoutDetails call)
+        payment_processor_account_id = options[:payment_processor_account_id] || :default
+        gateway                      = lookup_gateway(payment_processor_account_id, kb_tenant_id)
+        gw_response                  = gateway.details_for(token)
+        response, transaction        = save_response_and_transaction(gw_response, :details_for, kb_account_id, kb_tenant_id, payment_processor_account_id)
+        raise response.message unless response.success? and !response.payer_id.blank?
+
+        response.payer_id
+      end
+
       def add_required_options(kb_payment_transaction_id, kb_payment_method_id, context, options)
         payment_method = @payment_method_model.from_kb_payment_method_id(kb_payment_method_id, context.tenant_id)
 
-        options[:payer_id]     ||= payment_method.paypal_express_payer_id
-        options[:token]        ||= payment_method.paypal_express_token
-        options[:reference_id] ||= payment_method.token # baid
+        options[:payer_id]     ||= payment_method.paypal_express_payer_id if payment_method.paypal_express_payer_id
```

Comment:
You could simplify these lines with ActiveSupport's [presence](http://guides.rubyonrails.org/active_support_core_extensions.html#presence):

```
options[:payer_id] ||= payment_method.paypal_express_payer_id.presence
```